kernel/aster-nix/src/syscall/madvise.rs

```diff
@@ -17,23 +17,12 @@ pub fn sys_madvise(start: Vaddr, len: usize, behavior: i32) -> Result<SyscallRet
             let mut buffer = vec![0u8; len];
             read_bytes_from_user(start, &mut VmWriter::from(buffer.as_mut_slice()))?;
         }
-        MadviseBehavior::MADV_DONTNEED => madv_dontneed(start, len)?,
+        MadviseBehavior::MADV_DONTNEED => {}
```

Comment:
It's better to add a `warn!` here to clearly indicate that `MADV_DONTNEED` is not implemented. `MADV_DONTNEED` should _not_ be treated as a dummy for certain usages because the manual states that `MADV_DONTNEED` will impact the program semantics. Therefore, it is essential for the user to be aware that this implementation is dummy.

```plain
These advice values do not influence the semantics
of the application (except in the case of MADV_DONTNEED)
```